04_linked_list/lib/linked_list.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+const Node = require('./ll_node');
+const fs = require('fs');
+/********************************************************************************
+*         LINKED LIST MODULE                                                    *
+********************************************************************************/
+class LinkedList{
+  constructor(){
+    this.length = 1;
+    this.head = null;
+    this.current = null;
+    this.tail = null;
+    this.prev = null;

```

Comment:
`this.prev` is not truly implemented within your `append()` , `prepend()`, and other methods.
Either:
1. Truly implement `this.prev` to actually always have a correct value--previous node at all times or `null` if it's the head node. (This would change any time you append, prepend, reverse, manipulate the list.)
OR
2. Delete `this.prev`, and use a temporary variable `let prev = null;` that you manipulate in your methods like `reverse()` and `remove()`.